src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -634,6 +638,118 @@ public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, R
             }
         }
 
+        /// <summary>
+        /// Method to append InstanceAnnotations to the ODataResource and Property.
+        /// Instance annotations are annotations for a resource or a property and couldb be of contain a primitive, comple , enum or collection type 
+        /// These will be saved in to an Instance annotation dictionary
+        /// </summary>
+        /// <param name="resource">param for ODataResource</param>
+        /// <param name="resourceContext">param for ODataResourceContext</param>        
+        public virtual void AppendInstanceAnnotations(ODataResource resource, ResourceContext resourceContext)
+        {
+            IEdmStructuredType structuredType = resourceContext.StructuredType;
+            IEdmStructuredObject structuredObject = resourceContext.EdmObject;
+            PropertyInfo instanceAnnotationInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType,
+                resourceContext.EdmModel);
+
+            object value;
+            
+            if (instanceAnnotationInfo == null || structuredObject == null ||
+                !structuredObject.TryGetPropertyValue(instanceAnnotationInfo.Name, out value) || value == null)
+            {
+                return;
+            }
+
+            IODataInstanceAnnotationContainer instanceAnnotationContainer = value as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotationContainer != null)
+            {
+                IDictionary<string, object> clrAnnotations = instanceAnnotationContainer.GetAllAnnotations();
+
+                if (clrAnnotations != null)
+                {
+                    foreach (KeyValuePair<string, object> annotation in clrAnnotations)
+                    {
+                        AddOdataAnnotations(resource,null, resourceContext, annotation);
+                    }
+                }
+                                
+                foreach(ODataProperty property in resource.Properties)
+                {
+                    string propertyName = property.Name;
+
+                    if (property.InstanceAnnotations == null)
+                    {
+                        property.InstanceAnnotations = new List<ODataInstanceAnnotation>();
+                    }
+
+                    IDictionary<string, object> propertyAnnotations = instanceAnnotationContainer.GetAllPropertyAnnotation(propertyName);
+
+                    if (propertyAnnotations != null)
+                    {
+                        foreach (KeyValuePair<string, object> annotation in propertyAnnotations)
+                        {
+                            AddOdataAnnotations(null,property, resourceContext, annotation);
+                        }
+                    }                    
+                } 
+            }
+        }
+
+        private void AddOdataAnnotations(ODataResource resource, ODataProperty property, ResourceContext resourceContext, KeyValuePair<string, object> annotation)
```

Comment:
>ODataResource resource, ODataProperty property [](start = 41, length = 46)

Since both ODataResource and ODataProperty derive from ODataAnnotatable (which has the method you care about), replace these two arguments (one  of which is always false) with a single ODataAnnotatable annotatable parameter. Then you can remove the conditional logic in 722-729 and just call annotatable.InstanceAnnotations.Add(new ODataInstanceAnnotation(annotation.Key, annotationValue));